_drafts/building-a-cli-for-firmware-projects.md

```diff
@@ -0,0 +1,515 @@
+---
+title: Building a CLI for Firmware Projects using Invoke
+description:
+  "A walkthrough on how to build a command line interface (CLI) for a firmware
+  project using the Python Invoke package."
+author: tyler
+tags: [python]
+---
+
+<!-- excerpt start -->
+
+Building a small (or large) command line interface (CLI) for a project is a
+great way to get an entire team to build, test, debug, and work with a project
+in the same way using the same set of tools. This post goes into detail about
+how to think about a project's CLI and implementing one using the
+[Invoke](http://www.pyinvoke.org/) Python package.
+
+<!-- excerpt end -->
+
+## Why Build A Project CLI?
+
+By building a CLI for a project, it becomes a single way for any developer,
+script, or continuous integration system to perform the most common operations
+are performed everyday.
+
+For example, it helps convert launching JLinkGDBServer with the appropriate
```

Comment:
I'd say "for example, instead of having to remember that JLinkGDBServer must be invoked with"
[code]
I can just do
[code]